### PR TITLE
🌱 add clusterctl label to providers deployed with tilt

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
This PR adds clusterctl label to providers deployed with tilt, so commands like clusterctl move can work as expected

### CODE REVIEW

**Review by killianmuldoon — APPROVED** [review]
Thanks for this - working for me locally with tilt.

/lgmt

File: hack/tools/tilt-prepare/main.go [hunk 1]
```diff
@@ -668,7 +668,7 @@ func kustomizeTask(path, out string) taskFunction {
 // workloadTask generates a task for creating the component yaml for a workload and saving the output on a file.
 // NOTE: This task has several sub steps including running kustomize, envsubst, fixing components for debugging,
 // and adding the workload resource mimicking what clusterctl init does.
-func workloadTask(name, workloadType, binaryName, containerName string, ts *tiltSettings, path string, getAdditionalObjects func(string, []unstructured.Unstructured) (*unstructured.Unstructured, error)) taskFunction {
+func workloadTask(name, workloadType, binaryName, containerName string, ts *tiltSettings, path string, getAdditionalObject func(string, []unstructured.Unstructured) (*unstructured.Unstructured, error)) taskFunction {
```
Comment (sbueringer) [inline hack/tools/tilt-prepare/main.go:671]: I'm fine with the change, but what do you think about also setting the provider label?

Looks like `workloadTask` is only used to deploy the provider. So we could make the whole thing a bit more typed and also pass in the provider CRD as an actual Provider and take the manifest label from there

Comment (fabriziopandini) [reply hack/tools/tilt-prepare/main.go:671]: The provider label already exists on the component yaml, e.g
https://github.com/kubernetes-sigs/cluster-api/blob/320ec2f9101e99d431ee27e87abffdf224acf131/controlplane/kubeadm/config/default/kustomization.yaml#L6

Comment (sbueringer) [reply hack/tools/tilt-prepare/main.go:671]: Ah absolutely!

Just compared with clusterctl code but missed this part

**Pull request merged**